acme/api/middleware.go

```diff
@@ -324,6 +317,13 @@ func lookupJWK(next nextHTTP) nextHTTP {
 				render.Error(w, acme.NewError(acme.ErrorUnauthorizedType, "account is not active"))
 				return
 			}
+
+			if kid != acc.Location {
```

Comment:
It's a bit unclear what the values for `kid` and `Location` represent. In the first part, the `Location` seems to only contain the `new-account` URL; with the `kid` being the full key ID. But here it's compared directly, instead of the `Location` being suffixed with data. That suffix seems to be done in the storage layer. I think that shouldn't be done there for clarity.